### PR TITLE
Fix default installation without sidecar injection

### PR DESCRIPTION
**Pull request by guillaumeblaquiere**
The default installation doesn't work as described in the documentation.

I fixed it like this

 * manifest apply no longer exists in istioctl CLI.
 * Add runAsRoot to allow targetPort creation bellow 1024.

### CODE REVIEW

File: docs/install/installing-istio.md
Comment (JRBANCEL) [inline docs/install/installing-istio.md:94]: Why do we need this?
Last time I installed Istio (1.6.x), it opens ports in the 1024+ range and the service maps the real port to it.
For example, port 80 is mapped to 8080.
Is this not the case in 1.5?

Comment (guillaumeblaquiere) [reply docs/install/installing-istio.md:94]: I tested early in august, with version 1.6.x it worked (I prepared a demo for the GDG Cloud Melbourne). Recently, I tested the 1.7 version (for a demo to the GDG Nantes (a city in France), tonight: 6pm CEST) and it didn't work. I fixed like this. There might be a better/nicer way.

Comment (julz) [reply docs/install/installing-istio.md:94]: better than runAsRoot is to modify the configuration so that the requested `targetPorts` are high enough not to require root. Modifying the `ports` section of Istio-minimal-operator.yaml to look like the below should work with 1.7:

~~~~
            - name: status-port
              port: 15021
              targetPort: 15021
            - name: http2
              port: 80
              targetPort: 8080
            - name: https
              port: 443
              targetPort: 8443
~~~~

Comment (guillaumeblaquiere) [reply docs/install/installing-istio.md]: @julz, I committed a new version thank to your comment after having testing it!

**Comment by upodroid** [comment]
I spent several hours debugging around till I noticed cluster-local-gateway can't bind to priveleged ports.

Fresh knative install with istio 1.7

Can we get this merged ASAP?

**Comment by nak3** [comment]
@JRBANCEL @tcnghia @ZhiminXiang  I was also asked about the status of Istio 1.7 for Knative. I am wondering that we should merge this for 1.7 and put a note about the issue we have seen in CI? (I am happy to send a follow up PR.) We may also want to update istio-latest test to 1.7, though.

**Comment by tcnghia** [comment]
+1 on that.

I think at this point I'd go with testing 1.7 in `latest` with reduced parallelism in place. `stable` will still be older but at least we have some coverage.

**Comment by julz** [comment]
FWIW we've seen some big improvements in our load testing of knative with istio 1.7 with this fix applied: https://github.com/istio/istio/pull/26918. It's not in 1.7, but it should be in 1.7.1 in a few days. The tl;dr is there was a race when Endpoints get created before Service that can sometimes cause the istio endpoint cache never to get updated, and this situation seems to occur in knative semi-regularly under load. _Maybe_ this may even fix the problems at higher parallelism, if we're lucky, but either way - fyi.

**Comment by guillaumeblaquiere** [comment]
Any updates on this? Any helps/comments that I can bring to you?

**Comment by nak3** [comment]
I was thinking that we should wait for start testing Istio 1.7.1 which will include https://github.com/istio/istio/pull/26985 but maybe we don't need to wait for it?

/lgtm
/assign @tcnghia @ZhiminXiang @JRBANCEL 
/hold

**Comment by ZhiminXiang** [comment]
I think I may miss some context. What Istio versions could this PR apply to?

**Comment by nak3** [comment]
It depends on `istioctl` version which users use. But without this PR, installation of istio 1.7 always fail. So this PR is kind of applying to Istio 1.7.

**Comment by JRBANCEL** [comment]
I don't think we need to start testing 1.7 since with or without this change Istio until 1.6 included works and this unblock 1.7.

I don't see any risk or downside of this change (except that i might create some downtime but even Istio doesn't seem to care about this :disappointed:)

File: docs/install/installing-istio.md
Comment (howardjohn) [inline docs/install/installing-istio.md:96]: I think it's just `istioctl install`? Unless there is an alias I am not familiar with

Comment (nak3) [reply docs/install/installing-istio.md:96]: Yeah, it seems `istioctl manifest install` was forgotten to remove. I am trying to change to remove it https://github.com/istio/istio/pull/27277 so we should use `istioctl install` instead.

Comment (guillaumeblaquiere) [reply docs/install/installing-istio.md]: I removed the manifest in this PR

**Comment by tcnghia** [comment]
/unhold

(there is no objection to JR suggestion https://github.com/knative/docs/pull/2775#issuecomment-689997066 )

**Review by tcnghia — APPROVED** [review]
/lgtm

**Pull request merged**
